File: Week 08/id_586/leetcode_387.cc
class Solution {
public:
	int firstUniqChar(string s) {
		vector<int> v(26);

		for(int i = 0; i < s.length(); ++i)
			v[s[i]-'a']++;

		for(int i = 0; i < s.length(); ++i)
			if(v[s[i]-'a'] == 1)
				return i;
		return -1;
	}
};

File: Week 08/id_381/leetcode_387.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        cache = {}
        for c in s:
            if c in cache:
                cache[c] += 1
            else:
                cache[c] = 1
        for i in range(len(s)):
            if cache[s[i]] == 1:
                return i
        return -1
